manifests/simple/rule.pp

```diff
@@ -1,41 +1,26 @@
 # default values are being ignored for now
 define patterndb::simple::rule (
-  Variant[String, Array[String]] $patterns,
-  $ruleset,
-  $id = $title,
-  $provider = 'puppet',
-  $ruleclass = 'system',
-  $context_id = undef,
-  $_embedded = false,
-  $context_timeout = undef,
-  $context_scope = undef,
-  $order = '00',
-  $actions = [],
+  Array[String[1]] $patterns,
```

Comment:
While I agree on this, it is a backwards incompatible change, isn't it ?

Reply:
Yes it is, I had not anticipated this when started to work on this, but accepting both `Thing` and `Array[Thing]` makes the code less manageable and using a rigid structure is expected to be less error prone.

Let's update the PR title and labels accordingly!